### PR TITLE
Add a mechanism for snfoundry.toml file discovery

### PR DESCRIPTION
**Pull request by THenry14**
https://github.com/foundry-rs/starknet-foundry/issues/1620
This PR is a part of stack, that aims to replace Scarb.toml tool configuration with snfoundry.toml:

-> Add mechanism for snfoundry.toml discovery
-- Replace Scarb.toml based configuration with snfoundry.toml based one (https://github.com/foundry-rs/starknet-foundry/pull/1644)
-- Use the same profile for sncast and scarb (https://github.com/foundry-rs/starknet-foundry/pull/1646)
-- Update docs (https://github.com/foundry-rs/starknet-foundry/pull/1647)

## Introduced changes

- Add a mechanism for snfoundry.toml file discovery

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/src/helpers/configuration.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        .find(|path| fs::metadata(path.join(CONFIG_FILENAME)).is_ok())
+        .map(|path| path.join(CONFIG_FILENAME))
+        .ok_or_else(|| {
+            anyhow!(
+                "Failed to find snfoundry.toml - not found in current nor any parent directories"
+            )
+        })
+}
+
+pub fn find_config_file() -> Result<Utf8PathBuf> {
```
Comment (MaksymilianDemitraszek) [inline crates/sncast/src/helpers/configuration.rs:18]: And here we would could leave name as it is to abstract the config searching process.

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM

image

File: crates/sncast/src/helpers/configuration.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
+use super::constants::CONFIG_FILENAME;
+use anyhow::{anyhow, Result};
+use camino::Utf8PathBuf;
+use std::fs;
+
+pub fn find_config_file_relative_to(current_dir: &Utf8PathBuf) -> Result<Utf8PathBuf> {
```
[the remark this answers is not in the document]
Comment (THenry14) [reply crates/sncast/src/helpers/configuration.rs]: changed :+1:

**Review by ddoktorski — APPROVED** [review]
LGTM

**Pull request merged**